lib/libzfs/libzfs_pool.c

```diff
@@ -995,9 +999,10 @@ zpool_name_valid(libzfs_handle_t *hdl, boolean_t isopen, const char *pool)
 				    "multiple '@' and/or '#' delimiters in "
 				    "name"));
 				break;
-			case NAME_ERR_NO_AT:
+
+			default:
 				zfs_error_aux(hdl, dgettext(TEXT_DOMAIN,
-				    "permission set is missing '@'"));
+				    "(%d) not defined"), why);
```

Comment:
We need to keep the `NAME_ERR_NO_AT` case which is legitimate and add the `default` case to it.  This additional should be pushed upstream.